sklearn/utils/fixes.py

```diff
@@ -335,3 +335,20 @@ def __call__(self, *args, **keywords):
             return self.func(*args, **kwargs)
 else:
     from functools import partial
+
+
+if np_version < (1, 6, 2):
+    # Allow bincount to accept empty arrays
+    # https://github.com/numpy/numpy/commit/40f0844846a9d7665616b142407a3d74cb65a040
+    def bincount(x, weights=None, minlength=None):
+        x = np.asarray(x, dtype=np.intp)
```

Comment:
I would avoid the call to 'asarray' and simply use 'len(x)', which I believe should be faster.